tests/beaker_tests/cisco_interface/test_interface_L2.rb

```diff
@@ -0,0 +1,127 @@
+# rubocop:disable Style/FileName
+###############################################################################
+# Copyright (c) 2014-2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+#
+# See README-develop-beaker-scripts.md (Section: Test Script Variable Reference)
+# for information regarding:
+#  - test script general prequisites
+#  - command return codes
+#  - A description of the 'tests' hash and its usage
+#
+###############################################################################
+#
+# 'test_interface_L2' primarily tests layer 2 interface properties.
+#
+###############################################################################
+require File.expand_path('../interfacelib.rb', __FILE__)
+
+# Test hash top-level keys
+tests = {
+  agent:            agent,
+  master:           master,
+  intf_type:        'ethernet',
+  operating_system: 'nexus',
+  resource_name:    'cisco_interface',
+}
+
+# Find a usable interface for this test
+intf = find_interface(tests)
+
+# Test hash test cases
+tests[:default_access] = {
+  desc:               "1.1 Default 'access' Properties",
+  title_pattern:      intf,
+  code:               [0],
+  preclean_intf:      true,
+  sys_def_switchport: true,
+  sys_def_sw_shut:    true,
+  manifest_props:     {
+    shutdown:        'default',
+    switchport_mode: 'access',
```

Comment:
We probably left it out because we have to first set the mode to access, trunk, etc before we can test the associated properties for that mode. It might be worthwhile to add a default test to this at some point.